File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
        
    data.sort(key=lambda x:(x[col-1], -x[0]))
    
    S_i = 0
    for i in range(len(data)):
        if not row_begin-1 <= i <= row_end-1: continue
        
        tmp = 0
        for j in range(len(data[i])):
            tmp += data[i][j] % (i +1)            
        
        S_i = tmp ^ S_i        
    
    return S_i

File: 프로그래머스/2/147354. 테이블 해시 함수/테이블 해시 함수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;


void printData(vector<vector<int>> data)
{
    for(auto d : data)
    {
        for(auto i : d)   
        {
            cout<< i<<" ";
        }
        cout << "" << endl;
    }
}

int solution(vector<vector<int>> data, int col, int row_begin, int row_end) {
    sort(data.begin(), data.end(), [&col](vector<int> a, vector<int> b){
        if(a[col-1] != b[col-1]) return a[col-1] < b[col-1];
        return a[0] > b[0];
    });
    
    int S_i = 0;
    for(int i = 0; i < data.size(); i++)
    {
        if(!(row_begin-1 <= i && i <= row_end-1)) continue;
        int tmp = 0;
        
        for(int j = 0; j < data[i].size(); j++) 
            tmp += data[i][j] % (i+1);
        
        S_i = tmp ^ S_i;
    }
    
    return S_i;
}